doc: Convert builders/images/* to markdown

bobby285271 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Closes #105758 and closes #105761

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Converted xml to md and updated images.xml to reference it.

bobby285271 committed:
doc/builders/images/*.xml: Convert to markdown

SuperSandro2000 commented on doc/builders/images/appimagetools.section.md:
````diff
@@ -0,0 +1,72 @@
+
+Depending on the type of AppImage you're wrapping, you'll have to use `wrapType1` or `wrapType2`.
+
+```{=docbook}
````
Can you convert this, too?

bobby285271 replied:
Done, please take a look!

jtojnar reviewed (approved):
Diff looks okay to me.

bobby285271 commented:
Added converts of `snaptools.xml`.

I changed the title of section 14.4.3 and move the examples into the the markdown file during converting, so please take a look!

Thanks in advance for reviewing :-)

jtojnar commented on doc/builders/images/snaptools.section.md:
```diff
@@ -0,0 +1,71 @@
+## Build a Graphical Snap {#ssec-pkgs-snapTools-build-a-snap-firefox}
+
+Graphical programs require many more integrations with the host. This example uses Firefox as an example, because it is one of the most complicated programs we could package.
+
```
Maybe we should add the example anchors as inline anchors:

```suggestion

[]{#ex-snapTools-buildSnap-firefox}
```

jtojnar commented on doc/builders/images/snaptools.section.md:
````diff
@@ -0,0 +1,71 @@
+
+Graphical programs require many more integrations with the host. This example uses Firefox as an example, because it is one of the most complicated programs we could package.
+
+```nix
````
Or mark the code block directly.

```suggestion
``` {#ex-snapTools-buildSnap-firefox .nix}
```

jtojnar reviewed (commented):
Otherwise, the snap parts look fine as well.

jtojnar reviewed (approved):
The diff looks good to me.

jtojnar commented:
HTML looks okay too.
